streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/service/StreamCatalogService.java

```diff
@@ -2341,4 +2343,55 @@ private String getTopologyComponentBundleJarName (TopologyComponentBundle topolo
     private Collection<File> listFiles(List<QueryParam> queryParams) {
         return dao.find(File.NAME_SPACE, queryParams);
     }
+
+    public Collection<TopologyTestRunHistory> listTopologyTestRunHistory(Long topologyId) {
```

Comment:
We need to review previous test history for previous version as well to compare why test fails on current version. 
And if my understanding is right, same topology version should have same topology DAG so test result should be same except end-user changes expected output, which gives no benefit.